File: web-app/src/app/norm/entity/Klass.ts
import {Teacher} from './Teacher';

/**
 * 班级实体
 */
export class Klass {
  id: number;
  name: string;
  teacher: Teacher;

  /**
   * 构造函数
   * @param id id
   * @param name 名称
   * @param teacher 教师
   */
  constructor(id: number, name: string, teacher: Teacher) {
    this.id = id;
    this.name = name;
    this.teacher = teacher;
  }
}

File: api/src/main/java/com/mengyunzhi/springBootStudy/entity/Klass.java
package com.mengyunzhi.springBootStudy.entity;

import javax.persistence.*;

@Entity
public class Klass {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    private Teacher teacher;

    private String name;

    public Klass() {
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}
